src/ppx/application.cpp

```diff
@@ -680,6 +683,27 @@ void Application::DispatchShutdown()
 {
     Shutdown();
 
+    // Save the metrics report to disk
+    if (mSettings.useMetrics) {
+        // Build a unique name for the report
```

Comment:
Optional: Personally I think it would be more readable if this was wrapped in a helper SaveMetrics. But I will leave that up to you, perhaps we should also move it to the Metrics Manager?

Reply:
I added a function SaveMetricsReportToDisk.